fe2-android/app/src/main/java/com/github/dedis/student20_pop/model/data/LAORepository.java

```diff
@@ -61,744 +46,313 @@
-    Optional<Election> electionOption = lao.getElection(channel.split("/")[3]);
-    if (!electionOption.isPresent()) throw new IllegalArgumentException("the election should be present when receiving a result");
-    return electionOption.get();
-  }
-
-  /**
-   * Retrieves the Lao in a given channel
-   * @param channel the channel on which the Lao was created
-   * @return the Lao corresponding to this channel
-   */
-  private Lao getLaoByChannel(String channel) {
-    String[] split = channel.split("/");
-    return laoById.get("/root/" + split[2]).getLao();
+    // We've a Broadcast
+    handleBroadcast(this, genericMessage, unprocessed);
   }
 
   /**
-   * Checks that a given channel corresponds to a LAO channel, i.e /root/laoId
-   * @param channel the channel we want to check
-   * @return true if the channel is a lao channel, false otherwise
+   * Helper method that sends a StateLao message if we are the organizer
+   *
+   * @param lao       Lao of the message being signed
+   * @param msg       Object of type MessageGeneral representing the current message being signed
+   * @param messageId Base 64 URL encoded Id of the message to sign
+   * @param channel   Represents the channel on which to send the stateLao message
    */
-  private boolean isLaoChannel(String channel) {
-    return channel.split("/").length == 3;
-  }
-
-  private boolean handleElectionEnd(String channel) {
-    Lao lao = getLaoByChannel(channel);
-    Election election = getElectionByChannel(channel);
-    election.setEventState(CLOSED);
-    lao.updateElection(election.getId(), election);
-    return false;
-  }
-
-  private boolean handleCastVote(String channel, CastVote data, String senderPk, String messageId) {
-    Lao lao = getLaoByChannel(channel);
-    Election election = getElectionByChannel(channel);
-
-    //We ignore the vote iff the election is ended and the cast vote message was created after the end timestamp
-    if (election.getEndTimestamp() >= data.getCreation() || election.getState()!= CLOSED) {
-      /* We retrieve previous cast vote message stored for the given sender, and consider the new vote iff its creation
-      is after (hence preventing reordering attacks) */
-      Optional<String> previousMessageIdOption = election.getMessageMap().entrySet().stream()
-              .filter(entry -> senderPk.equals(entry.getValue())).map(Map.Entry::getKey).findFirst();
-      //If there is no previous message, or that this message is the last of all received messages, then we consider the votes
-      if (!previousMessageIdOption.isPresent() ||
-              ((CastVote) messageById.get(previousMessageIdOption.get()).getData()).getCreation() <= data.getCreation()) {
-        election.putVotesBySender(senderPk, data.getVotes());
-        election.putSenderByMessageId(senderPk, messageId);
-        lao.updateElection(election.getId(), election);
+  public void sendStateLao(Lao lao, MessageGeneral msg, String messageId, String channel) {
```

Comment:
why do we need to keep this in LAORepository ? 
I'm probably missing something very obvious here :)

Reply:
I thought about moving this method to the WitnessMessageHandler class because its only usage is in handleWitnessMessage. For that I would need to create getters for the keyset manager and Gson in the LAORepository or pass them as arguments on all handle methods. I am not sure what's best